src/base/gui/widget.h

```diff
@@ -15,18 +15,34 @@ class Widget
 public:
 	virtual ~Widget() = default;
 
-	virtual void onPointerDown(const PointerEvent &event) = 0;
-	virtual void onPointerUp(const PointerEvent &event) = 0;
-	virtual void onPointerLeave(const PointerEvent &event) = 0;
-	virtual void onPointerMove(const PointerEvent &event) = 0;
-	virtual void onWheel(double delta) = 0;
-	virtual void onKeyPress(const Key &key,
+	virtual void onPointerDown(
```

Comment:
Canvas parameter is out of context on the mouse event handlers for an abstract widget.
Anyway, this whole class could be thrown out, since we don't have multiple widget types anymore.

Reply:
When we throw out tooltips and mouse events, this should be not a problem.
Currently, we need the canvas because one widget can be drawn to multiple canvases. And widget can store information about Canvas.